src/VisualStudio/Core/Def/Implementation/UnusedReferences/Dialog/UnusedReferencesTableProvider.DataSource.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.UnusedReferences;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.UnusedReferences.Dialog
+{
+    internal partial class UnusedReferencesTableProvider
+    {
+        internal class UnusedReferencesDataSource : ITableDataSource
+        {
+            public string SourceTypeIdentifier => nameof(UnusedReferencesDataSource);
```

Comment:
Is this constant also being used in other places to match things up?

Reply:
Each ITableDataSource must provide an Identifier. I was also using this name as the identifier for the TableManager instance.